### PR TITLE
Strategic Merge Patch doc

### PR DESCRIPTION
**Pull request by mengqiy**
Pull Strategic Merge Patch out of API convention doc.
And refactor it.

Use the format what I discussed with @pwittrock offline.
More content need to be fill by @pwittrock in a followup PR.

### CODE REVIEW

File: contributors/devel/strategic-merge-patch.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+`replace` directive indicates that the element that contains it should be replaced instead of being merged.
+
+### Syntax
+
+`replace` directive is used in both patch with directive marker and go struct tags.
+
+Example usage in the patch:
+
+```
+$patch: replace
```
Comment (pwittrock) [inline contributors/devel/strategic-merge-patch.md]: Put the example in the context of an actual json struct.  This doesn't show demonstrate how to create a patch with this directive.

Comment (pwittrock) [inline contributors/devel/strategic-merge-patch.md]: Nevermind, I should have kept reading :P

File: contributors/devel/strategic-merge-patch.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+finalizers:
+  - a
+  - b
+  - c
+```
+
+To delete items "b" and "c" from the original finalizers, the patch will be:
+
+```yaml
+$deleteFromPrimitiveList/finalizers:
```
Comment (pwittrock) [inline contributors/devel/strategic-merge-patch.md]: Would you add the context around this like you did for the others?

**Comment by pwittrock** [comment]
Generally looks good.  Just one comment about adding an example for the delete for primitives

File: contributors/devel/strategic-merge-patch.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    image: nginx-1.0
+  - $patch: replace   # any further $patch operations nested in this list will be ignored
+```
+
+
+## `delete` Directive
+
+### Purpose
+
+`delete` directive indicates that the element that contains it should be deleted.
```
Comment (pwittrock) [inline contributors/devel/strategic-merge-patch.md]: Does this apply to maps, or just lists?  Will maps setting to null is how items are deleted right?

File: contributors/devel/strategic-merge-patch.md (hunk 2 above)
Comment (mengqiy) [inline contributors/devel/strategic-merge-patch.md]: Done.

File: contributors/devel/strategic-merge-patch.md (hunk 3 above)
Comment (mengqiy) [inline contributors/devel/strategic-merge-patch.md]: >Does this apply to maps, or just lists?

Both map and list of maps.

> Will maps setting to null is how items are deleted right?

That's one way. The other way is using delete directive in map.
I updated the doc to cover both.

File: contributors/devel/api-conventions.md [hunk 4]
```diff
@@ -555,172 +550,7 @@ below.
...
-To delete a field of a map:
-
-```yaml
-name: nginx
-image: nginx-1.0
-labels:
-  live: null  # set the value of the map key to null
-```
-
+Details of Strategic Merge Patch is in this [doc](../strategic-merge-patch.md).
```
Comment (pwittrock) [inline contributors/devel/api-conventions.md]: > Details of Strategic Merge Patch is in this [doc](../strategic-merge-patch.md).

How about:

Details of Strategic Merge Patch are covered [here](../strategic-merge-patch.md).

**Comment by pwittrock** [comment]
Looks good, one minor comment then ready to merge.

**Comment by mengqiy** [comment]
Addressed the comment. And squashed.

**Comment by pwittrock** [comment]
Thanks!

**Pull request merged**